script.js

```diff
@@ -1,12 +1,298 @@
-//You can edit ALL of the code here
+//Global Variables
+const episodeSearch = document.getElementsByClassName("season-names");
+const episodeCard = document.getElementsByClassName("card");
+const showCard = document.getElementsByClassName("show-card")
+const rootElem = document.getElementById("root");
+
+let inputSearch = document.createElement("INPUT");
+inputSearch.setAttribute("type", "search");
+
+let selector = document.createElement("select");
+selector.setAttribute("name", "episodes")
+selector.className = "selector";
+
+let shows = document.createElement("select");
+shows.setAttribute("name", "series")
+shows.className = "shows";
+
+let bod = document.getElementsByTagName("BODY")[0];
+let main = document.createElement("div")
+main.setAttribute("class", "main")
+bod.appendChild(main)
+
+let pages = document.createElement("div")
+pages.setAttribute("class", "pages")
+bod.appendChild(pages)
+
+
+let showId;
+let episode;
+
+//Setup Function
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  searchBar()
+  searchFilter()
+  showsList()
+  showSelection()
+  getShows()
+  citation()
+}
+
+//Populates the page with the shows
+function getShows(){
+  const rootElem = document.getElementById("root");
+
+  let shows = getAllShows();
+  let allShows = shows.sort(function compare(a, b) {
+    if (a.name.toUpperCase() < b.name.toLocaleUpperCase()) { return -1 }
+    if (a.name.toUpperCase() > b.name.toUpperCase()) { return 1 }
+    return 0;
+  })
+
+  for(i=0; i<allShows.length; i++){
+    let mainDiv = document.createElement("div")
+    mainDiv.setAttribute("class", "show-card")
+
+    let title = `${allShows[i].name} `
+    let h2 = document.createElement("h2")
+    h2.setAttribute("id", allShows[i].id)
+    h2.addEventListener("click", presentEpisodes)
+    h2.innerHTML = title
+
+    let page = `${allShows[i].summary}`
+    let p = document.createElement("p")
+    p.setAttribute("id", "show-text")
+    p.innerHTML = page
+
+    let status = `Genre: ${allShows[i].genres} Status: ${allShows[i].status}`
+    let h4 = document.createElement("h4")
+    h4.setAttribute("class", "status")
+    h4.innerHTML = status
+
+    let headers = `Rating: ${allShows[i].rating.average} Runtime: ${allShows[i].runtime}`
+    let h5 = document.createElement("h5")
+    h5.setAttribute("class", "headers")
+    h5.innerHTML = headers
+
+    let image = document.createElement("img")
+    image.setAttribute("class", "show-images")
+    let img = allShows[i].image?.medium
+    image.src = img
```

Comment:
again, really nicely structured - clean, concise code. Honestly you have no idea how much this ability separates the strong programmers from the average ones.